### PR TITLE
Multi merge mo edits

### PR DESCRIPTION
**Pull request by carvilsi**
after resolve conflicts… will be more conflicts on this merge…

### CODE REVIEW

**Comment by malamili** [comment]
- remove weaver dist file

File: src/CoreManager.coffee [hunk 1]
```diff
@@ -160,44 +174,51 @@ class CoreManager
...
+    payload = JSON.stringify(payload)
+    request.get("#{@endpoint}/file/downloadByID?payload=#{payload}")
+    .on('response', (res) ->
+      res
+    )
+
   GET: (path, payload, target) ->
     @REQUEST("GET", path, payload, target)
 
+  GET_HTTP: (path, payload, target) ->
```
Comment (malamili) [inline src/CoreManager.coffee]: remove

File: src/Weaver.coffee [hunk 2]
```diff
@@ -23,10 +23,8 @@ class Weaver
     @ACL         = require('./WeaverACL')
     @Role        = require('./WeaverRole')
     @User        = require('./WeaverUser')
-    if window?
-      @FileBrowser      = require('./WeaverFileBrowser')
-    else
-      @File      = require('./WeaverFile')
+    if !window?
+      @File        = require('./WeaverFile') # avoiding problems on browsers trying to load node's fs stuff
```
Comment (malamili) [inline src/Weaver.coffee]: Is this If still needed? Now we dont have WeaverFile in browsers

File: src/WeaverFile.coffee [hunk 3]
```diff
@@ -3,74 +3,46 @@ writeFile        = require('fs-writefile-promise')
 Error            = require('./Error')
 WeaverError      = require('./WeaverError')
 readFile         = require('fs-readfile-promise')
+fs               = require('fs')
+CoreManager      = Weaver.getCoreManager()
 
 
-class WeaverFile
+class WeaverFile extends Weaver.Node
```
Comment (malamili) [inline src/WeaverFile.coffee]: Should not extend Node anymore

File: src/WeaverNode.coffee [hunk 4]
```diff
@@ -16,12 +16,12 @@ class WeaverNode
     # All operations that need to get saved
     @pendingWrites = [Operation.Node(@).create()]
 
+
   # Node loading from server
   @load: (nodeId, target, Constructor) ->
-
     Constructor = WeaverNode if not Constructor?
 
-    new Weaver.Query(target).get(nodeId, Constructor)
```
Comment (malamili) [inline src/WeaverNode.coffee]: Why is Constructor gone?

File: src/Weaver.coffee (hunk 2 above)
Comment (carvilsi) [inline src/Weaver.coffee]: We cannot use node fs in the browser and it's causing errors

File: src/WeaverNode.coffee (hunk 4 above)
Comment (carvilsi) [inline src/WeaverNode.coffee]: I do not know this change is not mine

**Pull request merged**